### PR TITLE
[AERIE-1651] Integrate `aerie_scheduler` database with scheduler

### PR DESCRIPTION
**Pull request by pcrosemurgy**
* **Tickets addressed:** AERIE-1651
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

- Adds `aerie_sched` database to deployed and local Docker stack.
  - All tables are required to have a `scheduling_` prefix for now until the Hasura issue described below is resolved or documented.
- Updates Hasura Docker image to v2.2.0
  - Encountered duplicate table name in Hasura: https://github.com/hasura/graphql-engine/issues/7895
  - When using v2.2.0 the issue is still present. The closing comment for the PR above does not offer an explanation of how  namespaces work or should be used.
  - https://github.com/hasura/graphql-engine/issues/6974 implements namespaces, but it's still not clear how to use them without an example or user-facing documentation.
- Connects scheduler app driver to `aerie_sched` DB and `AERIE_SCHED_*` environment variables.
- Renames `*SCHED_*` env. variables to `*SCHEDULER_*`.

## Verification
- Tested with local inspection of SQL tables in `tablePlus`.
- Tested with local inspection of Hasura through locally-deployed instance.
- All tests passing in `scheduler-server`.

## Documentation
None.

## Future work
@kroffo will implement `PostgresSpecificationRepository` and `PostgresResultsCellRepository`. @adrienmaillard will work on integrating those with the existing scheduler framework.

### CODE REVIEW

File: deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql
Comment (pcrosemurgy) [inline deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql]: This was lifted from @kroffo's initial tables for the scheduler. I've commented out this section since it refers to the `mission_model` table. It's expected that Kenny will uncomment and fix this as part of his work.

File: deployment/postgres-init-db/sql/sched/tables/scheduling_uploaded_file.sql
Comment (pcrosemurgy) [inline deployment/postgres-init-db/sql/sched/tables/scheduling_uploaded_file.sql]: This table is maybe not needed for the scheduler server, but this exists to show that providing a custom prefix to `uploaded_file` works around Hasura's duplicate table metadata issue. This table can be removed as part of this PR if desired, for now it serves as an example of our Hasura workaround.

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/PostgresResultsCellRepository.java
Comment (adrienmaillard) [inline scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/PostgresResultsCellRepository.java]: rename `planId` to `specificationId`  in this file.

**Review by adrienmaillard — COMMENTED** [review]
This looks good to me. Nothing to say except renaming some variables.

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/SchedulerAppDriver.java
Comment (mattdailis) [inline scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/SchedulerAppDriver.java]: :small_red_triangle_down: Similar to Adrien's comment - this variable has the word plan in it, but should probably say specification instead.

File: deployment/postgres-init-db/init-aerie.sh
Comment (patkenneally) [inline deployment/postgres-init-db/init-aerie.sh]: :large_blue_circle: Is there are reason we dropped the "uler" off aerie_sched(uler) here? What happened Paul? I thought you and I both loved more verbose naming over shortening. 😆

File: deployment/postgres-init-db/init-aerie.sh
Comment (patkenneally) [inline deployment/postgres-init-db/init-aerie.sh]: :large_blue_circle: Same here (sched) 😄 . Happy with whatever you choose though. Just asking the question.

File: deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql
Comment (patkenneally) [reply deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql]: I think if Kenny is to add/implement this, then it should first appear along with the code that implements it. So we can remove it entirely from this PR.

Separately, given the scheduler db and the merlin db are separated by an Aerie module boundary (different components) we won't be able to refer to any merlin db tables. Is this intended to be literal or more of a reminder that we will need to implement a mechanism to manage the mission model id to which a scheduling rule will be associated?

File: deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql
Comment (patkenneally) [inline deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql]: We can just have "A rule for ..." you seem to use "rule" everywhere else.

File: deployment/postgres-init-db/sql/sched/tables/scheduling_uploaded_file.sql
Comment (patkenneally) [reply deployment/postgres-init-db/sql/sched/tables/scheduling_uploaded_file.sql]: What do you think of pulling this out into a separate branch for now? We can hold it there as an example until we determine if we need to use this approach.

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/mocks/InMemoryResultsCellRepository.java
Comment (patkenneally) [inline scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/mocks/InMemoryResultsCellRepository.java]: :small_red_triangle_down: specificationId

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/mocks/InMemoryResultsCellRepository.java
Comment (patkenneally) [inline scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/mocks/InMemoryResultsCellRepository.java]: :small_red_triangle_down: specificationId

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

Nice work Paul. Some speedy work here. This scheduler is coming together nicely.

File: deployment/postgres-init-db/init-aerie.sh
Comment (pcrosemurgy) [reply deployment/postgres-init-db/init-aerie.sh]: lol normally I'd use `aerie_scheduler` here but this was just to be consistent with the `docker-compose.yaml`'s name for the DB (committed by Steve some time ago I think):
https://github.com/NASA-AMMOS/aerie/blob/181e751451dbc2fd0f1068d54d745d7c68e4f8e4/deployment/docker-compose.yml#L45

File: deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql
Comment (pcrosemurgy) [reply deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql]: Sounds good, I'll remove this :) And it was meant to be a reminder that the `aerie_sched` DB and `aerie_merlin` DB are not to be coupled so that `constraint` would need to be rewritten, removed, or reworked somehow.

File: deployment/postgres-init-db/init-aerie.sh
Comment (patkenneally) [reply deployment/postgres-init-db/init-aerie.sh]: Ahh true. That's fair, I prefer the longer version, but yes I follow your reasoning. If you want to change it I'd support that 😃, but no pressure.

File: deployment/postgres-init-db/sql/sched/tables/scheduling_uploaded_file.sql
Comment (pcrosemurgy) [reply deployment/postgres-init-db/sql/sched/tables/scheduling_uploaded_file.sql]: That's fair, I'll take it out of this PR.

File: deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql
Comment (patkenneally) [reply deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql]: Wonderful. And thanks for sharing the context!

File: deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql
Comment (pcrosemurgy) [reply deployment/postgres-init-db/sql/sched/tables/scheduling_rule.sql]: Good catch! Will do

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/SchedulerAppDriver.java
Comment (pcrosemurgy) [reply scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/SchedulerAppDriver.java]: Just pushed this change, good catch!

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/PostgresResultsCellRepository.java
Comment (pcrosemurgy) [reply scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/PostgresResultsCellRepository.java]: Done, thanks for catching this!

**Review by patkenneally — APPROVED** [review]

File: deployment/postgres-init-db/init-aerie.sh
Comment (pcrosemurgy) [reply deployment/postgres-init-db/init-aerie.sh]: Actually, I'll tag this on as a commit to this PR. Would be nice to keep this commit separate from the first 2 commits though, that way the name change and the root of the PR are semantically separated.

Comment (patkenneally) [reply deployment/postgres-init-db/init-aerie.sh]: 🥳. Agreed.

Comment (pcrosemurgy) [reply deployment/postgres-init-db/init-aerie.sh]: Done :) Also renamed some env. vars.

**Review by kroffo — APPROVED** [review]
Assuming I'll be adding the remaining database tables in my branch, looks good to me!

**Review by camargo — APPROVED** [review]
Great work. LGTM.

**Pull request merged**
